nntrainer/include/layer.h

```diff
@@ -402,6 +409,11 @@ class Layer {
   WeightDecayParam weight_decay;
 
   WeightIniType weight_ini_type;
+
+  /**
+   * @brief     Gradient for the weights in this layer
+   */
+  std::vector<std::reference_wrapper<Tensor>> gradients;
```

Comment:
Not related to this PR though, how about moving `std::vector<Tensor> weights` out here so we can access to the current weights from out side as well?

Reply:
good suggestion, will do in a separate PR, and add corresponding getter.